### PR TITLE
doc: Add more details to the guide on publishing plugins

### PR DESCRIPTION
**Pull request by pilvikala**
#### Summary

Adding a bit more detail to the guide on publishing plugins - what the team name/display name is, and how to use the plugin visibility and release stage for staging plugin releases.

### CODE REVIEW

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx [hunk 1]
```diff
@@ -14,19 +14,35 @@ Publishing to the CloudQuery Hub is only supported for Go plugins at the moment.
...
 - You have created a [CloudQuery Cloud](https://cloud.cloudquery.io/) account and completed the onboarding process to create a team
-- You have created the plugin you'd like to publish on https://cloud.cloudquery.io/ under the relevant team
 - You have the [CloudQuery CLI](/docs/quickstart) installed (version >= `v3.27.1`)
 - The plugin you'd like to publish is written in Go, and uses an SDK version >= `v4.17.1`
 - The plugin you'd like to publish is initialized using the plugin's name, team and kind. See example [here](https://github.com/cloudquery/cloudquery/blob/b7ef6f6ed8948272a429f35614fa28559397227a/plugins/source/test/resources/plugin/plugin.go#L15)
 - You are authenticated to [CloudQuery Cloud](https://cloud.cloudquery.io/) using the `cloudquery login` command
 
+## Create a Plugin Definition on Hub
+
+Before publishing a plugin, you need to create a plugin definition on Hub.
```
Comment (erezrokah) [inline website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx:24]:
```suggestion
Before publishing a plugin, you need to create a plugin definition on the CloudQuery Cloud site.
```

Not sure how we are officially referring to the Cloud site, but it's different from the Hub

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx [hunk 2]
```diff
@@ -14,19 +14,35 @@ Publishing to the CloudQuery Hub is only supported for Go plugins at the moment.
...
 - You have the [CloudQuery CLI](/docs/quickstart) installed (version >= `v3.27.1`)
 - The plugin you'd like to publish is written in Go, and uses an SDK version >= `v4.17.1`
 - The plugin you'd like to publish is initialized using the plugin's name, team and kind. See example [here](https://github.com/cloudquery/cloudquery/blob/b7ef6f6ed8948272a429f35614fa28559397227a/plugins/source/test/resources/plugin/plugin.go#L15)
 - You are authenticated to [CloudQuery Cloud](https://cloud.cloudquery.io/) using the `cloudquery login` command
 
+## Create a Plugin Definition on Hub
+
+Before publishing a plugin, you need to create a plugin definition on Hub. 
+
+Log in to https://cloud.cloudquery.io/. If you have not created a team yet, you will be asked to create a new one. You can also create a new team to publish the plugins with.
```
Comment (erezrokah) [inline website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx:26]:
```suggestion
Log in to https://cloud.cloudquery.io/. If you have not created a team yet, you will be asked to create a new one.
```

I'm not sure I understand the second part of the sentence. Could we remove it for brevity or change it to "You can use the created team to publish plugins"

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx [hunk 3]
```diff
@@ -14,19 +14,35 @@ Publishing to the CloudQuery Hub is only supported for Go plugins at the moment.
...
+
+<img src="/images/docs/publishing-plugins/create-team.png" width="50%" style={{margin:"1em"}} />
+
+The display name will be visible on Hub next to your plugin. The team name is going to be used in configs to reference your plugin using `team-name/plugin-name`.
```
Comment (erezrokah) [inline website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx:30]:
```suggestion
The display name will be visible on Hub next to your plugin. The team name is going to be used in configurations to reference your plugin using `<team-name>/<plugin-name>`.
```

We often use `<>` to denote a value the user should replace

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx [hunk 4]
```diff
@@ -14,19 +14,35 @@ Publishing to the CloudQuery Hub is only supported for Go plugins at the moment.
...
-5. Once you're ready run `cloudquery plugin publish -f` to publish a non draft version of the plugin. This version will be immutable and will show up in <https://hub.cloudquery.io/>. Allow up to 1 hour for the Hub to reflect the changes
+1. (Optional, recommended) In the root directory of your plugin repository run `git tag v1.0.0` to tag the version you're about to publish (replace `v1.0.0` with the version you'd like to publish).
+2. (Optional, recommended) Run `git push origin v1.0.0` to push the tag.
+3. Run `go run main.go package --docs-dir docs -m 'feat: Initial release' v1.0.0 .` to package the plugin. `v1.0.0` should match the tag you created in step 1. The `-m` specifies the changelog message that will be used in the release notes and it supports markdown. See example [here](https://hub.cloudquery.io/plugins/.source/cloudquery/alicloud/v4.0.14/versions). `docs` should be a directory containing markdown files that serve as documentation for the plugin. Read more about the documentation format [here](#documentation-format).
```
Comment (erezrokah) [inline website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx:43]:
```suggestion
3. Run `go run main.go package --docs-dir docs -m 'feat: Initial release' v1.0.0 .` to package the plugin. `v1.0.0` should match the tag you created in step 1. The `-m` specifies the changelog message that will be used in the release notes and it supports markdown. See example [here](https://hub.cloudquery.io/plugins/source/cloudquery/alicloud/v4.0.14/versions). `docs` should be a directory containing markdown files that serve as documentation for the plugin. Read more about the documentation format [here](#documentation-format).
```

I believe this dot breaks the link

**Review by erezrokah — CHANGES_REQUESTED** [review]
Great change 🚀 Added a few comments.

Also can we keep the file as `.md`. Not sure why we need `.mdx`. `.md` supports HTML too

**Comment by pilvikala** [comment]
> Great change 🚀 Added a few comments.
> 
> Also can we keep the file as `.md`. Not sure why we need `.mdx`. `.md` supports HTML too

How do I add the HTML to markdown? The html `` does not get rendered when the file is `.md`.

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx (hunk 3 above)
Comment (pilvikala) [reply website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx]: fixed

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx (hunk 4 above)
Comment (pilvikala) [reply website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx]: fixed

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx [hunk 5]
```diff
@@ -14,19 +14,35 @@ Publishing to the CloudQuery Hub is only supported for Go plugins at the moment.
...
+4. Run `cloudquery plugin publish` to publish a draft version of the plugin. The version will show up under the versions tab of your plugin in (https://cloud.cloudquery.io). As long as the version is in draft it's mutable and you can re-package the plugin and publish it again.
```
[the remark this answers is not in the document]
Comment (pilvikala) [reply website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx]: fixed so it's not bare

File: website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx [hunk 6]
```diff
@@ -14,19 +14,35 @@ Publishing to the CloudQuery Hub is only supported for Go plugins at the moment.
...
 
+## Create a Plugin Definition on Hub
+
+Before publishing a plugin, you need to create a plugin definition on the CloudQuery Cloud site.
+
+Log in to [CloudQuery Cloud](https://cloud.cloudquery.io/). If you have not created a team yet, you will be asked to create a new one.
+
+<img src="/images/docs/publishing-plugins/create-team.png" width="50%" style={{margin:"1em"}} />
+
+The display name will be visible on Hub next to your plugin. The team name is going to be used in configurations to reference your plugin using `<team-name>/<plugin-name>`.
```
Comment (disq) [inline website/pages/docs/developers/publishing-a-plugin-to-the-hub.mdx:30]: it's technically `<team-name>/<plugin-kind>/<plugin-name>`, no?
```suggestion
The display name will be visible on Hub next to your plugin. The team name is going to be used in configurations to reference your plugin using `<team-name>/<plugin-kind>/<plugin-name>`.
```

**Review by disq — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Pull request merged**